tests/conftest.py

```diff
@@ -178,3 +179,90 @@ def custom_template(tmp_dir, dvc):
+
+
+@pytest.fixture(autouse=True)
+def isolate(tmp_path_factory, monkeypatch: pytest.MonkeyPatch):
+    """
+    Isolate our tests so that things like global configuration files and the
+    like do not affect our test results.
+    We use an autouse function scoped fixture because we want to ensure that
+    every test has it's own isolated home directory.
+    """
+
+    # Create a directory to use as our home location.
+    path = tmp_path_factory.mktemp("mock_fs")
+    home_dir = os.path.join(str(path), "home")
+    os.makedirs(home_dir)
+
+    # Create a directory to use as a fake root
+    fake_root = os.path.join(str(path), "fake-root")
+    os.makedirs(fake_root)
+
+    if sys.platform == "win32":
+        # Note: this will only take effect in subprocesses...
+        home_drive, home_path = os.path.splitdrive(home_dir)
+        monkeypatch.setenv("USERPROFILE", home_dir)
+        monkeypatch.setenv("HOMEDRIVE", home_drive)
+        monkeypatch.setenv("HOMEPATH", home_path)
+        for env_var, sub_path in (
+            ("APPDATA", "AppData/Roaming"),
+            ("LOCALAPPDATA", "AppData/Local"),
+        ):
+            path = os.path.join(home_dir, *sub_path.split("/"))
+            monkeypatch.setenv(env_var, path)
+            os.makedirs(path)
+    else:
+        # Set our home directory to our temporary directory, this should force
+        # all of our relative configuration files to be read from here instead
+        # of the user's actual $HOME directory.
+        monkeypatch.setenv("HOME", home_dir)
+        # Isolate ourselves from XDG directories
+        monkeypatch.setenv(
+            "XDG_DATA_HOME",
+            os.path.join(
+                home_dir,
+                ".local",
+                "share",
+            ),
+        )
+        monkeypatch.setenv(
+            "XDG_CONFIG_HOME",
+            os.path.join(
+                home_dir,
+                ".config",
+            ),
+        )
+        monkeypatch.setenv("XDG_CACHE_HOME", os.path.join(home_dir, ".cache"))
+        monkeypatch.setenv(
+            "XDG_RUNTIME_DIR",
+            os.path.join(
+                home_dir,
+                ".runtime",
+            ),
+        )
+        monkeypatch.setenv(
+            "XDG_DATA_DIRS",
+            os.pathsep.join(
+                [
+                    os.path.join(fake_root, "usr", "local", "share"),
+                    os.path.join(fake_root, "usr", "share"),
+                ]
+            ),
+        )
+        monkeypatch.setenv(
+            "XDG_CONFIG_DIRS",
+            os.path.join(
+                fake_root,
+                "etc",
+                "xdg",
+            ),
+        )
+
+    monkeypatch.setenv("GIT_CONFIG_NOSYSTEM", "1")
+
+    with open(os.path.join(home_dir, ".gitconfig"), "wb") as fp:
+        fp.write(
+            b"[user]\n\tname = DVC Tester\n\temail = dvctester@example.com\n"
+            b"[init]\n\tdefaultBranch = master\n"
```

Comment:
When I do `git init`, it prints the following warning:
```console
$ git init
hint: Using 'master' as the name for the initial branch. This default branch name
hint: is subject to change.
```

So, I don't think we should depend on the hardcoded default branch in our tests **in the future**. To do so, we should introduce `scm.default_branch` property and use that in the tests instead.